Prevent homedir removal on upgrade

rafamanzo opened the pull request:
The upgrade install both versions side by side and just after the
install of the new one removes the old. This is triggering the
postinstall script which was removing the homedir.

Both RPM and deb pass arguments to the script that permit to distinguish
upgrades from removals. But they do this differently forcing a more
complex conditional.

For further information check:
- On RPM - https://docs.fedoraproject.org/en-US/Fedora_Draft_Documentation/0.1/html/RPM_Guide/ch09s04s05.html
- On deb - https://wiki.debian.org/MaintainerScripts#Upgrading

rafamanzo committed:
Prevent homedir removal on upgrade

The upgrade install both versions side by side and just after the
install of the new one removes the old. This is triggering the
postinstall script which was removing the homedir.

Both RPM and deb pass arguments to the script that permit to distinguish
upgrades from removals. But they do this differently forcing a more
complex conditional.

For further information check:

* On RPM - https://docs.fedoraproject.org/en-US/Fedora_Draft_Documentation/0.1/html/RPM_Guide/ch09s04s05.html
* On deb - https://wiki.debian.org/MaintainerScripts#Upgrading

diegoamc commented:
What about increasing release numbers? Will we do it after merging into master?

rafamanzo commented:
I think so.

diegoamc commented:
It hasn't worked for me on CentOS. The homedir is still being removed. I've tested the kalibro-configurations package.

It has worked for the same package on Debian.

rafamanzo commented:
Yes, I'm working on some debug over CentOS right now :)

Hope for the best

rafamanzo committed:
Fix postuninstall check for upgrade on CentOS

RPM does not work as we expected, actually passing as argument 0 for
removals and 1 for upgrades.

rafamanzo commented:
What the referenced fedora's doc does not mention is that this rule (and that is mentioned by http://www.ibm.com/developerworks/library/l-rpm2/) applies only for `%pre` and `%pos` not for `%preun` and `%posun` which is our target.

So the proper value for `$1` is 0 for removal and 1 for upgrade.

The relevant fix has been added to this PR.

diegoamc commented:
It seems to work fine now. Great job! :+1:
